CMakeLists.txt

```diff
@@ -244,7 +265,11 @@ if(NOT UA_COMPILE_AS_CXX AND (CMAKE_COMPILER_IS_GNUCC OR "x${CMAKE_C_COMPILER_ID
                     -Wnested-externs
                     -Wmultichar
                     -Wundef
-                    -Wc++-compat)
+                    -Wc++-compat
+                    -fno-strict-aliasing # fewer compiler assumptions about pointer types
+                    -fexceptions # recommended for multi-threaded C code, also in combination with C++ code
+                    -fstack-protector-strong # stack protector - should be default anyway
```

Comment:
I remember that there was some issue with the stack protector and using the nodeset compiler.
See:
https://github.com/open62541/open62541/blob/master/CMakeLists.txt#L653

When using the stack protector and the big NS0 file, gcc tries to optimize the file and takes more than 30 seconds compared to currently 1-2 seconds.
Since it is disabled explicitly for the ua_namespace0.c file, it should be fine to set it here.

Reply:
Ah ok nice! I recognized that the strong stackprotector is available only since gcc 4.9 but the travis-ci version is 4.8.4. I might add a check if it is available and activate the "default" protector otherwise (-fstack-protector).